src/orders/components/OrderPayment/utils.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  OrderDetails_order,
+  OrderDetails_order_giftCards_events
+} from "@saleor/orders/types/OrderDetails";
+import { GiftCardEventsEnum } from "@saleor/types/globalTypes";
+
+export const extractOrderGiftCardUsedAmount = (
+  order: OrderDetails_order
+): number | undefined => {
+  if (!order) {
+    return undefined;
+  }
+
+  const { id, giftCards } = order;
+
+  const orderRelatedEvent = giftCards.reduce((resultEvent, { events }) => {
```

Comment:
You're always iterating through an entire array and it seems unnecessary. Why not use `find` instead, it will automatically end the execution when the nested `events.find` finds a value and returns it.